markdown/extensions/tables.py

```diff
@@ -85,7 +85,39 @@ def _split_row(self, row, border):
                 row = row[1:]
             if row.endswith('|'):
                 row = row[:-1]
-        return row.split('|')
+        return self._split(row, '|')
+
+    def _split(self, row, marker):
+        """ split a row of text with some code into a list of cells. """
+        if self._row_has_code(row):
+            ## fallback on old behaviour
+            return row.split(marker)
+        escaped = False
+        elements = []
+        current = ''
+        for i in row:
+            if i == '`' and (current == '' or current[-1] != '\\'):
+                ## Found an un-escaped backtick
+                escaped = not escaped
+
+            if i == marker and not escaped:
+                elements.append(current)
+                current = ''
+            else: 
+                current += i
+        if escaped:
+            ## This should never happen
+            return row.split(marker)
+        elements.append(current)
+        return elements
+
+    def _row_has_code(self, row):
+        count_total_backtick = row.count('`')
+        count_escaped_backtick = row.count('\`')
+        count_backtick = count_total_backtick - count_escaped_backtick
+        ## odd number of backticks, 
+        ## we won't be able to build correct code blocks
+        return (count_backtick % 2 != 0)
```

Comment:
Just a suggestion: `return count_backtick & 1` is a more beautiful form of writing this.